sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -130,6 +172,156 @@ public HttpPipeline getHttpPipeline() {
+    {
+        return getDigitalTwinWithResponse(digitalTwinId, clazz)
+            .map(DigitalTwinsResponse::getValue);
+    }
+
+    /**
+     * Gets a digital twin.
+     *
+     * @param digitalTwinId The Id of the digital twin. The Id is unique within the service and case sensitive.
+     * @param clazz The model class to convert the response to.
+     * @param <T> The generic type to convert the response to.
+     * @return A Http response containing the application/json digital twin
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public <T> Mono<DigitalTwinsResponse<T>> getDigitalTwinWithResponse(String digitalTwinId, Class<T> clazz)
+    {
+        return withContext(context -> getDigitalTwinWithResponse(digitalTwinId, clazz, context));
+    }
+
+    <T> Mono<DigitalTwinsResponse<T>> getDigitalTwinWithResponse(String digitalTwinId, Class<T> clazz, Context context) {
+        return protocolLayer
+            .getDigitalTwins()
+            .getByIdWithResponseAsync(digitalTwinId, context)
+            .map(response -> {
+                T genericResponse = mapper.convertValue(response.getValue(), clazz);
+                DigitalTwinsResponseHeaders twinHeaders = mapper.convertValue(response.getDeserializedHeaders(), DigitalTwinsResponseHeaders.class);
+                return new DigitalTwinsResponse<>(response.getRequest(), response.getStatusCode(), response.getHeaders(), genericResponse, twinHeaders);
+            });
+    }
+
+    /**
+     * Updates a digital twin.
+     *
+     * @param digitalTwinId The Id of the digital twin.
+     * @param digitalTwinUpdateOperations The application/json-patch+json operations to be performed on the specified digital twin
+     * @return an empty Mono
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> updateDigitalTwin(String digitalTwinId, List<Object> digitalTwinUpdateOperations)
+    {
+        return updateDigitalTwinWithResponse(digitalTwinId, digitalTwinUpdateOperations, new RequestOptions())
+            .flatMap(voidResponse -> Mono.empty());
+    }
+
+    /**
+     * Updates a digital twin.
+     *
+     * @param digitalTwinId The Id of the digital twin.
+     * @param digitalTwinUpdateOperations The application/json-patch+json operations to be performed on the specified digital twin
+     * @param options The optional settings for this request
+     * @return A Http response
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DigitalTwinsResponse<Void>> updateDigitalTwinWithResponse(String digitalTwinId, List<Object> digitalTwinUpdateOperations, RequestOptions options)
+    {
+        return withContext(context -> updateDigitalTwinWithResponse(digitalTwinId, digitalTwinUpdateOperations, options, context));
+    }
+
+    Mono<DigitalTwinsResponse<Void>> updateDigitalTwinWithResponse(String digitalTwinId, List<Object> digitalTwinUpdateOperations, RequestOptions options, Context context) {
```

Comment:
Aren't these return values supposed to be `void` with a lowercase 'v'?

Reply:
Not in this case, no. 
Here we want the `<T>` in `Response<T>` to be the class representation of the lower case `void`, which is upper case `Void`.
This would be similar to the async API returning `Mono<Integer>`, which the sync version returns an `int`.

Reply:
I guess I don't get it then.

https://stackoverflow.com/questions/15538219/java-lang-void-vs-void-vs-null#:~:text=void%20is%20a%20return%20type%20signifying%20no%20return.,rare%20instance%20where%20you%20need%20it%20for%20reflection.

Reply:
Seems every response in that stackoverflow thread had a different take on it.

Reply:
The `void` keyword signifies that there is nothing returned from this method call. The type `Void` is the class representation of that -> it cannot be instantiated, so it cannot have any values. 
In scenarios where we have to represent the class representation of `void`, we use `Void`. This is useful in scenarios where the method definition defines the return type to be a generic class `<T>`, and we cannot modify it to return nothing -> `void`. (eg: when implementing methods from interfaces that we do not control, etc).